client/my-sites/upgrades/index.js

```diff
@@ -276,8 +276,8 @@ module.exports = function() {
 
 	if ( config.isEnabled( 'upgrades/checkout' ) ) {
 		page(
-			'/checkout/thank-you',
-			upgradesController.redirectIfThemePurchased,
+			'/checkout/:site/:receipt_id/thank-you',
```

Comment:
This was already mentioned before but we should also use `:receiptId` instead of `:receipt_id`.